nixjs-rt/src/errors/typeError.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ErrorMessage,
+  err,
+  NixError,
+  instanceToClass,
+  classListToErrorMessage,
+} from ".";
+import { NixTypeClass, NixTypeInstance } from "../lib";
+
+export class NixTypeMismatchError {
+  constructor(
+    public readonly expected: NixTypeClass[],
+    public readonly got: NixTypeClass,
+  ) {}
+
+  toDefaultErrorMessage(): ErrorMessage {
+    return err`Expected ${classListToErrorMessage(this.expected)}, but got ${this.got}`;
```

Comment:
I guess with the simpler API proposed above it wouldn't look all that worse:
```suggestion
    return err`Expected ${boldAnyOf(this.expected)}, but got ${bold(this.got)}`;
```